### PR TITLE
Install getting started

### LINKED ISSUE #969: Install directory and release should contain getting_started directory
New users should be able to provision a VM and build their app against the release. Currently the need to clone to get the VM provisioning scripts in `getting_started` (or roll their own...)

### PR DESCRIPTION
**Pull request by achamayou**
fix #969

### CODE REVIEW

File: getting_started/create_vm/make_vm.sh [hunk 1]
```diff
@@ -2,19 +2,25 @@
...
 DEFAULT_PUBLIC_KEY_PATH=~/.ssh/id_rsa.pub
-PUBLIC_KEY_PATH=${1:-${DEFAULT_PUBLIC_KEY_PATH}}
+PUBLIC_KEY_PATH=${2:-${DEFAULT_PUBLIC_KEY_PATH}}
 
 az login
-az group create --name "$RG" --location "$REGION" --subscription "$SUBSCRIPTION"
+az group create --name "$RG" --location "$REGION" --subscription "$1"
 oe-engine generate --api-model "$CFG" --ssh-public-key "$PUBLIC_KEY_PATH" --output-directory generated
-az group deployment create --name "$RG-$REGION" --resource-group "$RG" --template-file generated/azuredeploy.json --parameters @generated/azuredeploy.parameters.json --subscription "$SUBSCRIPTION"
+az deployment group create --name "$RG-$REGION" --resource-group "$RG" --template-file generated/azuredeploy.json --parameters @generated/azuredeploy.parameters.json --subscription "$1"
```
Comment (achamayou) [inline getting_started/create_vm/make_vm.sh]: `group deployment` is deprecated and has been replaced by `deployment group` (yes, really).

File: getting_started/create_vm/make_vm.sh [hunk 2]
```diff
@@ -2,19 +2,25 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the Apache 2.0 License.
 
-set -ex
+set -e
+
+if [ "$#" -lt 1 ]; then
```
Comment (achamayou) [inline getting_started/create_vm/make_vm.sh]: Passing explicit args rather than env var.

**Comment by jumaffre** [comment]
As I understand it, the expected workflow is:
1. Download the CCF install on your local machine (e.g. laptop). Untar.
2. Install Azure CLI.
3. Run the scripts in `getting_started/create_vm` to create a new DC VM.
4. SSH onto the VM and download the CCF install _again_. Untar.
5. Run the ansible scripts in `getting_started/setup_vm`
(Alternative to 5. is to run the ansible playbooks directly from the machine on which 1. was run, provided that ansible is installed there).
6. Build you app.

All files apart from `getting_started/create_vm` in the first install are not actually required but I guess it's the best way to distribute this script. An alternative would be to package `getting_started/create_vm` and export it as a release artifact. What do you think?

**Comment by achamayou** [comment]
2. is done by ./pre_make_vm.sh, so it's sort part of 3.?
4. well yeah...

I think I prefer to keep things simple for now with a single SDK tarball that has All You Need (tm), but arguably the create_vm stuff could migrate elsewhere, since it's not actually CCF-specific (oe_engine? openenclave-azure-get-started?) and we could just reference it.

I suspect this is a post-0.9 thing though.

File: getting_started/create_vm/make_vm.sh [hunk 3]
```diff
@@ -2,19 +2,25 @@
...
+  echo "make_vm.sh takes at least an Azure subscription name,"
```
Comment (jumaffre) [inline getting_started/create_vm/make_vm.sh]: `echo "$0 takes at least an Azure subscription name,"`?

File: getting_started/create_vm/make_vm.sh [hunk 4]
```diff
@@ -2,19 +2,25 @@
...
+  echo "and an optional path to an SSH public key"
+  exit 1
+fi
 
 # Azure Resource Group
-RG=ccf-dev
+RG=ccf-dev-rg
 # Region, only eastus and westeurope are supported at the moment
```
Comment (jumaffre) [inline getting_started/create_vm/make_vm.sh]: Update comment

File: getting_started/create_vm/destroy_vm.sh [hunk 5]
```diff
@@ -2,16 +2,21 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the Apache 2.0 License.
 
-set -ex
+set -e
 
-read -p "Are you sure? " -n 1 -r
+if [ "$#" -ne 1 ]; then
+  echo "make_vm.sh takes an Azure subscription name"
```
Comment (eddyashton) [inline getting_started/create_vm/destroy_vm.sh]: This says `make_vm.sh`, but is in `destroy_vm.sh`. Think this is easier to read as a usage hint, ie:

```
echo "Usage: $0 SUBSCRIPTION_NAME"
```

File: getting_started/create_vm/make_vm.sh (hunk 3 above)
Comment (eddyashton) [inline getting_started/create_vm/make_vm.sh]: As above, would prefer `"Usage: $0 SUBSCRIPTION_NAME [path/to/ssh/public/key]"`

File: getting_started/create_vm/vm.json [hunk 6]
```diff
@@ -2,20 +2,21 @@
...
+        "name": "ccf-dev-vm",
         "osType": "Linux",
-        "vmSize": "Standard_DC4s",
-        "ports": [22, 25000],
+        "vmSize": "Standard_DC8_v2",
+        "ports": [22],
         "hasDNSName": true,
-        "isVanilla": true
+        "isVanilla": true,
+	      "osDiskType": "StandardSSD_LRS"
```
Comment (eddyashton) [inline getting_started/create_vm/vm.json]: Nit: inconsistent whitespace.

File: getting_started/create_vm/make_vm.sh (hunk 3 above)
Comment (achamayou) [reply getting_started/create_vm/make_vm.sh]: Indeed, thank you!

Comment (achamayou) [reply getting_started/create_vm/make_vm.sh]: Yes

File: getting_started/create_vm/vm.json (hunk 6 above)
Comment (achamayou) [reply getting_started/create_vm/vm.json]: fixed

File: getting_started/create_vm/make_vm.sh (hunk 3 above)
Comment (achamayou) [reply getting_started/create_vm/make_vm.sh]: fixed

File: getting_started/create_vm/destroy_vm.sh (hunk 5 above)
Comment (achamayou) [reply getting_started/create_vm/destroy_vm.sh]: yes

File: getting_started/create_vm/make_vm.sh (hunk 4 above)
Comment (achamayou) [reply getting_started/create_vm/make_vm.sh]: done

**Comment by achamayou** [comment]
CI breaks for unrelated reasons (quote verification issue in ACC)

**Pull request merged**